lib/takeaway.rb

```diff
@@ -0,0 +1,80 @@
+require_relative 'order'
+require_relative 'menu'
+require_relative 'twilio'
+
+class Takeaway
+
+  def initialize(order = Order.new, text_client = TwilioClient.new)
+    @order = order
+    @order_done = false
+    @text_client = text_client
+  end
+
+  def take_order
+    puts "Welcome to Su's takeaway"
+    take_number unless @order_done == true
+    show_menu unless @order_done == true
+    edit_order until @order_done == true
+    cost
+  end
+
+  def place_order
+    fail "You have not created an order yet" unless @order_done == true
+    confirmation_text
+    reset_order
+  end
+
+  def cancel_order
+    @order = Order.new
+    @order_done = false
+    puts "Your order has been cancelled."
+  end
+
+  def show_menu
+    @order.show_menu
+  end
+
+  def cost 
+    @order.total_cost
+  end
+
+  private
+
+  def take_number
+    puts "Please enter your 10 digit mobile number"
+    @number = STDIN.gets.chomp
+  end
+
```

Comment:
edit_order and order_dish methods are slightly long but still very readable due to the clear puts statements showing you where you are up to in the method